src/ui/control/navigation_control.js

```diff
@@ -102,6 +102,37 @@ class NavigationControl {
             this._handler = new DragRotateHandler(map, {button: 'left', element: this._compass});
             DOM.addEventListener(this._compass, 'mousedown', this._handler.onMouseDown);
             DOM.addEventListener(this._compass, 'touchstart', this._handler.onMouseDown, {passive: false});
+            DOM.addEventListener(this._compass, 'keydown', (e) => {
+                const bearingStep = 15,
```

Comment:
I've made the implementation here since the zoom in/out button clicks are performed here, and the KeyboardHandler seems focused on handling keyboard events on the map, not the NavigationControl. This does lead to these constants defined twice, is that okay or should they be defined once and them imported into both the KeyboardHandler and NavigationControl?

Reply:
Sorry, which constants are defined twice here?

We don't have a clean separation of focus and input handling, probably KeyboardHandler is a bit misnamed if we have multiple controls that need to handle the keyboard but KeyboardHandler just represents the focused map.

Reply:
bearingStep and ptichStep are also defined at https://github.com/mapbox/mapbox-gl-js/blob/237713463802baa397db21dce14d8987b8e36e66/src/ui/handler/keyboard.js#L8-L9

Reply:
Ah thank you. I think it's good to have them defined once and imported. I don't think we'd want to modify these values separately. Maybe it's a good time to start a `src/ui/constants.js`? I am having a hard time finding other examples in the project of a shared constants file but it seems reasonable to me.